### ISSUE TITLE
Watches use separate queues which could cause race conditions in Reconcile()

### ISSUE THREAD

**Issue by McCodeman** [kubernetes-incubator/apiserver-builder-alpha#245]
**TL;DR**
Watches enqueue calls to `Reconcile()` on a separate queue than the primary controller work queue. This could allow simultaneous updates to a resource causing a race condition in the `Reconcile()` function. Is there a reason that the event handlers associated with a watch (informer) would not enqueue onto the controllers primary work queue in order to serialize state changes?

**Scenario**
Suppose we have the following resources: Foo, Bar, and Baz.

Foo is responsible for creating and owning Bars and Bazs. In the Foo controller, a watch is created for Bars and Bazs. These watches cause Foos with matching Ownership to be reconciled.

We now have 3 queues calling `Reconcile()` on Foo (each related resource watch queue and the primary  resource queue). Decisions made in reconcile may not have a complete view of the state or may override state modified by the other queues since reconciliation of Foo is not serialized.

Watch function that creates additional workqueues:
https://github.com/kubernetes-incubator/apiserver-builder/blob/14201e804a58a140011a0044b15331f477535f91/pkg/controller/informers.go#L67L78

This setup is documented at:
https://github.com/kubernetes-incubator/apiserver-builder/blob/master/docs/watching_kubernetes_resources.md
